### PR TITLE
create init bucket in user's project

### PR DESCRIPTION
**Pull request by MatthewBemis**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala [hunk 1]
```diff
@@ -39,19 +39,29 @@ class BucketHelper(dataprocConfig: DataprocConfig,
     * Creates the dataproc init bucket and sets the necessary ACLs.
     */
   def createInitBucket(googleProject: GoogleProject, bucketName: GcsBucketName, serviceAccountInfo: ServiceAccountInfo): Future[GcsBucketName] = {
+    def flatMapList(entities: List[GcsEntity], role: GcsRole): Future[Unit] = {
+      entities match {
+        case Nil => Future.successful(())
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala]: fyi, there's `Future.unit`

**Comment by MatthewBemis** [comment]
jenkins retest

**Comment by MatthewBemis** [comment]
jenkins retest

**Comment by MatthewBemis** [comment]
jenkins retest

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala [hunk 2]
```diff
@@ -39,19 +39,29 @@ class BucketHelper(dataprocConfig: DataprocConfig,
...
       leoEntity = userEntity(serviceAccountProvider.getLeoServiceAccountAndKey._1)
       // When we receive a large number (e.g. 200) of simultaneous cluster creation requests,
       // we hit Google bucket creation/deletion request quota of one per approx. two seconds.
       // Therefore, we are adding a second layer of retries on top of the one existing within
       // the googleStorageDAO.createBucket method
       _ <- retryUntilSuccessOrTimeout(failureLogMessage = s"Init bucket creation failed for Google project '$googleProject'")(30 seconds, 5 minutes) { () =>
-        googleStorageDAO.createBucket(dataprocConfig.leoGoogleProject, bucketName, bucketSAs, List(leoEntity))
+        googleStorageDAO.createBucket(googleProject, bucketName, bucketSAs, List(leoEntity))
       }
+      _ <- flatMapList(List(leoEntity) ++ bucketSAs, GcsRoles.Reader)
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala]: I think the call to `createBucket` will set the default object acl already. So is this call really needed?

See: https://github.com/broadinstitute/workbench-libs/blob/develop/google/src/main/scala/org/broadinstitute/dsde/workbench/google/HttpGoogleStorageDAO.scala#L56

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala [hunk 3]
```diff
@@ -39,19 +39,29 @@ class BucketHelper(dataprocConfig: DataprocConfig,
...
       // Leo service account -> Owner
       // available service accounts ((cluster or default SA) and notebook SA, if they exist) -> Reader
+      // Default object ACLs for all of the above mentioned SAs will be Reader //todo: is reader the right thing?
       bucketSAs <- getBucketSAs(googleProject, serviceAccountInfo)
       leoEntity = userEntity(serviceAccountProvider.getLeoServiceAccountAndKey._1)
       // When we receive a large number (e.g. 200) of simultaneous cluster creation requests,
       // we hit Google bucket creation/deletion request quota of one per approx. two seconds.
       // Therefore, we are adding a second layer of retries on top of the one existing within
       // the googleStorageDAO.createBucket method
       _ <- retryUntilSuccessOrTimeout(failureLogMessage = s"Init bucket creation failed for Google project '$googleProject'")(30 seconds, 5 minutes) { () =>
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala]: Maybe we should remove this retry (and above comment) -- it was added explicitly to deal with rate limiting due to init buckets created in the same project. The user bucket creation does not have a retry here. Note `GoogleStorageDAO` is already retrying internally, so we will still have retries (just not retries on top of retries :)).

**Comment by MatthewBemis** [comment]
jenkins retest

**Comment by rtitle** [comment]
@MatthewBemis are you actively working on this? If not, mind if I pick it up? As expected this is coming up in perf testing: https://docs.google.com/document/d/1__vT45PlPMewNwg7L3Wi-w_vFup9jnjdX3zPCntQxCQ/edit?usp=gmail_thread&ts=5d03e924

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Pull request merged**